partitioned-table.md

```diff
@@ -753,78 +751,153 @@ In addition, there are limitations on the compatibility of `EXCHANGE PARTITION`
 
 ```sql
-ALTER TABLE members TRUNCATE PARTITION p1;
+ALTER TABLE members ADD PARTITION (PARTITION p1990 VALUES LESS THAN (2000));
 ```
 
 ```
-Query OK, 0 rows affected (0.03 sec)
+ERROR 1493 (HY000): VALUES LESS THAN value must be strictly increasing for each partition
 ```
 
-> **Note:**
+#### Reorganize partition
+> **Warning:**
 >
-> `ALTER TABLE ... REORGANIZE PARTITION` is currently unsupported in TiDB.
+> This is an experimental feature, which might be changed or removed without prior notice. The syntax and implementation may change before GA. If you find a bug, please open an issue in the [TiDB repository](https://github.com/pingcap/tidb/issues). 
 
-Add a partition:
+Split a partition:
+{{< copyable "sql" >}}
 
+```sql
+ALTER TABLE members REORGANIZE PARTITION `p1990to2010` INTO
+(PARTITION p1990 VALUES LESS THAN (2000),
+ PARTITION p2000 VALUES LESS THAN (2010),
+ PARTITION p2010 VALUES LESS THAN (2020),
+ PARTITION p2020 VALUES LESS THAN (2030),
+ PARTITION pMax VALUES LESS THAN (MAXVALUE));
+
+ALTER TABLE member_level REORGANIZE PARTITION l5_6 INTO
+(PARTITION l5 VALUES IN (5),
+ PARTITION l6 VALUES IN (6));
+```
+
+Merge partitions:
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE members REORGANIZE PARTITION pBefore1950,p1950 INTO (PARTITION pBefore1960 VALUES LESS THAN (1960));
+
+ALTER TABLE member_level REORGANIZE PARTITION l1,l2 INTO (PARTITION l1_2 VALUES IN (1,2));
+```
+
+Change the whole partitioning scheme:
 {{< copyable "sql" >}}
 
 ```sql
-ALTER TABLE members ADD PARTITION (PARTITION p3 VALUES LESS THAN (2010));
+ALTER TABLE members REORGANIZE PARTITION pBefore1960,p1960,p1970,p1980,p1990,p2000,p2010,p2020,pMax INTO
+(PARTITION p1800 VALUES LESS THAN (1900),
+ PARTITION p1900 VALUES LESS THAN (2000),
+ PARTITION p2000 VALUES LESS THAN (2100));
+
+ALTER TABLE member_level REORGANIZE PARTITION l1_2,l3,l4,l5,l6 INTO
+(PARTITION lOdd VALUES IN (1,3,5),
+ PARTITION lEven VALUES IN (2,4,6));
+```
+
+Reorganize partition (including merging or splitting partitions) in the generic form changes a list of partitions into a new set of partition definitions. It cannot change the type of partitioning (LIST or RANGE [COLUMNS]). For RANGE [COLUMNS] partitioned tables, the list of partitions needs to be in a single range. One can only change the end of the range if it includes the last partition. If the end is changed and there exists rows that no longer fits, the DDL will fail with an error.
+
+Range partitions needs to be a single range:
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE members REORGANIZE PARTITION p1800,p2000 INTO (PARTITION p2000 VALUES LESS THAN (2100));
 ```
 
-When partitioning tables by Range, `ADD PARTITION` can be only appended to the very end of a partition list. If it is appended to an existing Range partition, an error is reported:
+```
+ERROR 8200 (HY000): Unsupported REORGANIZE PARTITION of RANGE; not a single range
+```
 
+New partition definitions that does not accommodate all existing rows:
```

Comment:
```suggestion
New partition definitions must match the same range or set of values as the original definitions.
```

Reply:
New partitions can have a different range or list of different values, but the existing rows in the reorganized partitions must fit into the new partitions. Not sure how to express that.

Reply:
How about "New partition definitions must cover the existing range or set of values so that existing rows can fit into the new partitions."?